### ISSUE TITLE
Cop idea: Array Coercion

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12334]
## Description

Use `Array()` instead of explicit `Array` check or `[*var]`, when dealing with a variable you want to treat as an Array, but you’re not certain it’s an array.

https://rubystyle.guide/#array-coercion

## Behavior

```ruby
# bad
paths = [paths] unless paths.is_a?(Array)
paths.each { |path| do_something(path) }

# bad (always creates a new Array instance)
[*paths].each { |path| do_something(path) }

# good (and a bit more readable)
Array(paths).each { |path| do_something(path) }
```

**Comment by fatkodima** [comment]
There is already a `Style/ArrayCoercion` cop for this.

**Comment by ydakuka** [comment]
Hm, I've just checked. rubocop finds the offenses if the `only` option will pass.

```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/models/test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/models/test.rb --only Style/ArrayCoercion
Inspecting 1 file
C

Offenses:

app/models/test.rb:5:5: C: [Correctable] Style/ArrayCoercion: Use Array(paths) instead of explicit Array check.
    paths = [paths] unless paths.is_a?(Array)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

**Comment by fatkodima** [comment]
Yeah, that's because it is disabled by default - https://github.com/rubocop/rubocop/blob/c1d0505452116f3a5275e1a5fb94f244af72a801/config/default.yml#L3090-L3097

**Comment by ydakuka** [comment]
I see. Thank you.
